### PR TITLE
added linux-4k PKGBUILD

### PR DESCRIPTION
**Pull request by Sadoon-AlBader**
This PR adds a PKGBUILD for linux-4k, a package for the linux kernel configured with 4K memory pages instead of the default 64K.

Check this out for some good info:
https://docs.voidlinux-ppc.org/faq/index.html#why-use-4-kib-page-kernels-instead-of-64-kib-like-other-distros

### CODE REVIEW

**Review by kth5 — CHANGES_REQUESTED** [review]

File: linux-4k/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,206 @@
+# POWER Maintainer: Alexander Baldeck <alex.bldck@gmail.com>
+# Maintainer: Jan Alexander Steffens (heftig) <jan.steffens@gmail.com>
+
+pkgbase=linux
```
Comment (kth5) [inline linux-4k/PKGBUILD:4]: This should also be linux-4k to not conflict with the default linux package.

File: linux-4k/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,206 @@
+# POWER Maintainer: Alexander Baldeck <alex.bldck@gmail.com>
```
[the remark this answers is not in the document]
Comment (Sadoon-AlBader) [reply linux-4k/PKGBUILD]: Oops. fixed lol

File: linux-4k/PKGBUILD (hunk 1 above)
Comment (Sadoon-AlBader) [reply linux-4k/PKGBUILD]: I thought I already did that o_o
Done

File: linux-4k/config [hunk 3]
```diff
@@ -0,0 +1,9046 @@
...
+CONFIG_USE_PERCPU_NUMA_NODE_ID=y
+CONFIG_HAVE_MEMORYLESS_NODES=y
+CONFIG_ARCH_SELECT_MEMORY_MODEL=y
+CONFIG_ARCH_SPARSEMEM_ENABLE=y
+CONFIG_ARCH_SPARSEMEM_DEFAULT=y
+CONFIG_SYS_SUPPORTS_HUGETLBFS=y
+CONFIG_ILLEGAL_POINTER_VALUE=0x5deadbeef0000000
+CONFIG_ARCH_MEMORY_PROBE=y
+CONFIG_NODES_SPAN_OTHER_NODES=y
+CONFIG_PPC_4K_PAGES=y
```
[the remark this answers is not in the document]
Comment (Sadoon-AlBader) [reply linux-4k/config:444]: It's 100% compatible on my end, no crashes and every software I've used works.
I also tried 64k pages on Void as opposed to the default 4k, all works fine.
The only thing that has issues is hangover and btrfs afaik.
I think even if you personally don't use 4k pages it's nice to offer an option and the best part is that it's a simple pkgname change and one config line change :)

[the remark this answers is not in the document]
Comment (kth5) [reply linux-4k/config:444]: Yeah, I'm just using my box as a daily driver so I'm concerned to change something that has potential to break many things. I'm currently building linux-4k with you modifications and will boot it up for a while to see how it performs.

[the remark this answers is not in the document]
Comment (Sadoon-AlBader) [reply linux-4k/config:444]: If it breaks anything on my end I'll let you know.
Keep in mind you can't boot a 64kb btrfs partition on a 4k kernel.

**Pull request merged**